### PR TITLE
Search

### PR DESCRIPTION
**Pull request by georgebrock**
Adds <kbd>/</kbd>, <kbd>?</kbd>, <kbd>n</kbd> and <kbd>N</kbd>.

Includes a refactor of arguments: Rather than operations knowing an argument type ("literal", "operation", "none") and using that to determine what to do with key presses, now there are argument objects (`LiteralArgument`, `CommandLineArgument`, `NullArgument` and `Operation` which has a compatible interface with arguments) and the operation just passed the key presses on for the argument to handle.

### CODE REVIEW

File: js/base.js [hunk 1]
```diff
@@ -258,26 +273,54 @@
...
+    };
+
+    Vimulator.Base.prototype.moveToNext = function (target, options) {
+        var position = this.findNext(target, options);
+        if (position) {
+            this.moveCursor(position.row, position.col);
+            return true;
+        } else {
+            return false;
+        }
```
Comment (mike-burns) [inline js/base.js]: This pattern!

... that's all.

**Comment by mike-burns** [comment]
Would a better name for `NullArgument` be `NoArgument`?

**Comment by georgebrock** [comment]
Good call, I've renamed it to `NoArgument`.

File: js/base.js (hunk 1 above)
Comment (georgebrock) [inline js/base.js]: I've updated this:
- `moveCursor` now accepts an object, so I can pass the result of `findNext` directly to it
- `findNext` returns either `{row: …, col: …, found: true}` or `{found: false}`, never `null`

**Comment by georgebrock** [comment]
Ready for another review, @mike-burns

**Comment by mike-burns** [comment]
I like the refactoring. No further suggestions.

**Comment by georgebrock** [comment]
Thanks Mike, merging.

**Pull request merged**